### PR TITLE
[Bugfix][CI][Hardware][AMD] Fix various e4m3fn -> e4m3fnuz normalization issues

### PR DESCRIPTION
**Pull request by mawong-amd**
## Purpose
This PR fixes various issues in e4m3fnuz handling encountered by various quantization code paths (ModelOpt, online). In particular, on `e4m3fnuz` platforms, `normalize_e4m3fn_to_e4m3fnuz` should be called exactly once on a given FP8 linear layer after weights are loaded.
However,
- Some PyTorch linear kernel-based code paths used during ModelOpt quantization were not calling this function.
- Some code paths used in online quantization methods were trying to call this function twice, once in the linear layer and once more in the linear layer's kernel implementation.

The end result is that the Quantization test group should now be passing on ROCm.

This PR also removes a Quark mxfp4 model that was causing failures in the Kernels MoE test group because it is incompatible with HF Transformers v5.
## Test Plan
`VLLM_TEST_FORCE_LOAD_FORMAT=auto pytest -v -s quantization/ --ignore quantization/test_blackwell_moe.py`
`pytest -v -s kernels/moe --ignore=kernels/moe/test_modular_oai_triton_moe.py`
These tests should be run as part of the Quantization and Kernels MoE test groups in vLLM AMD CI.

## Test Result
The above test groups pass, and there are no regressions.

cc @kenroche and @AndreasKaratzas

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

File: vllm/model_executor/layers/quantization/modelopt.py [hunk 1]
```diff
@@ -726,8 +726,7 @@ def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
 
         layer.weight_scale = Parameter(scale.contiguous(), requires_grad=False)
 
-        if hasattr(self, "fp8_linear"):
-            self.fp8_linear.process_weights_after_loading(layer)
+        self.w8a8_block_fp8_linear.process_weights_after_loading(layer)
```
[the remark this answers is not in the document]
Comment (mawong-amd) [reply vllm/model_executor/layers/quantization/modelopt.py:731]: This shouldn't be a concern.
Firstly, the original `hasattr(self, "fp8_linear")` check seems to be dead code because `fp8_linear` is never set anywhere inside this class.
Secondly, even if we assume that it is a typo, i.e. that the attribute to check should be `"w8a8_block_linear"` rather than `"fp8_linear"`, this is initialized by `init_fp8_linear_kernel` which returns either a `FP8ScaledMMLinearKernel` or a `Fp8BlockScaledMMLinearKernel`, both of which have `process_weights_after_loading` defined.

**Comment by AndreasKaratzas** [comment]
@mawong-amd Just for the record, mention the reason behind the removal of the llama model, and also provide some explanation of purpose of the new `process_weights_after_loading` function (why is it necessary to have a new function/what is the motivation).

File: tests/kernels/moe/test_ocp_mx_moe.py [hunk 2]
```diff
@@ -64,7 +64,6 @@ def enable_pickle(monkeypatch):
     [
         ModelCase("fxmarty/qwen_1.5-moe-a2.7b-mxfp4", tp=2),
         ModelCase("fxmarty/deepseek_r1_3_layers_mxfp4", tp=8),
-        ModelCase("fxmarty/Llama-4-Scout-17B-16E-Instruct-2-layers-mxfp4", tp=1),
```
Comment (rasmith) [inline tests/kernels/moe/test_ocp_mx_moe.py:86]: We should probably either have the quark model redone or make it possible for vLLM to run this quark model correctly.
